modules/core/04-channel/keeper/packet.go

```diff
@@ -136,7 +136,8 @@ func (k Keeper) RecvPacket(
 		return errorsmod.Wrap(types.ErrChannelNotFound, packet.GetDestChannel())
 	}
 
-	if channel.State != types.OPEN {
+	counterpartyLastPacketSent, _ := k.GetCounterpartyLastPacketSequence(ctx, packet.GetDestPort(), packet.GetDestChannel())
+	if channel.State != types.OPEN && (channel.FlushStatus != types.FLUSHING || counterpartyLastPacketSent > packet.GetSequence()) {
```

Comment:
I find it easier to reason about like this:

```suggestion
	if !(channel.State == types.OPEN || (channel.FlushStatus == types.FLUSHING &&  packet.GetSequence() <= counterpartyLastPacketSent)) {
```

Reply:
we typically go with the approach of negating the spec conditions right? I do agree with you though that this makes it easier to reason (and, as a result, implement)

Reply:
agree with @crodriguezvega it's easier to reason about this way.

Reply:
should we actually add a condition in the code that if the channel is CLOSED then we can't recv packets? kinda weird if we can receive packets then right?

Reply:
we shouldn't be able to. If state is CLOSED, flush status should be NOTINFLUSH and the condition should resolve to true.

Reply:
oh! didnt see that comment, thanks! yea. that makes sense.